nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -348,6 +348,9 @@ def apply(
                 all_weight_params.append(weight_params)
                 weight_names.add(weight_name)
 
+        # Sort weight params to start compression with the bigger constants. This lowers peak memory footprint.
+        all_weight_params = sorted(all_weight_params, key=lambda wp: wp.num_weights, reverse=True)
```

Comment:
1) Mixed precision relies on the order of weight params. I didn't find the place where it can be broken with this reording, but please double-check that with ratio=0.8 in the tests the same layers are chosen.

2) It does break the logic with ratio defining params, because it handles the last layer separately.

Reply:
@ljaljushkin thanks! As I understand it will be fine if I move reordering a few lines lower: e974befd226d22067f9ef89929213534dfaf1ec3. Right?

Reply:
We decided to move the sorting right before weights compression not to interfere with other algorithms